### ISSUE TITLE
Ruby Client Hard to Follow

### ISSUE THREAD

**Issue by jenglert** [gilt/apidoc#91]
TBH, I found the ruby client very hard to follow.  (For reference, I was using the client generated by transactional email service https://gist.github.com/jenglert/3539880a906da1cd1357).  Basically, we need to make it easier to figure out how to use the thing.  Some suggestions:

* It would greatly aid readability if the actual bits you might call were at the top of the file.
* Classes that are repeated in every client should be at the bottom, and hopefully below a comment indicating that stuff below the comment isn't client specific (and can be ignored by people looking for the interface).
* The models module should come after the clients module
* My method to send email was named 'post'.  This isn't really great in my opinion.  The method should be named sendMail IMO.  Consider someone reading an application of the client.  It would be much easier to parse
```
client.messages.sendMail(...)
```
rather than
```
client.messages.post(...)
```
* The method post accepted a hash instead of a list of parameters.  This makes it hard to know which information you need to pass in the Hash to get an email to send.  It also makes it hard to type check anything other than the fact that I supplied a hash.  For example, the following should have provided a more concrete error message:
```
>> client.messages.post({})
TransactionalEmailDeliveryService::HttpClient::ServerError: TransactionalEmailDeliveryService::HttpClient::ServerError
	from (irb):235:in `http_request'
	from (irb):204:in `do_request'
	from (irb):158:in `post'
	from (irb):81:in `post'
	from (irb):467
	from :0
```
* Error messages are fairly undecipherable.  For example, I have no idea why this is failing.  I think we should consider error messaging a first class product of the client.
```
>> client.messages.post({"idempotent_token"=>"test", "subject"=>"hi", "body"=>"body", "to"=>"englert.james@gmail.com", "account"=> "gilt", "email_description"=>"jimtest", "priority"=>1})
TransactionalEmailDeliveryService::HttpClient::ServerError: TransactionalEmailDeliveryService::HttpClient::ServerError
	from (irb):235:in `http_request'
	from (irb):204:in `do_request'
	from (irb):158:in `post'
	from (irb):81:in `post'
	from (irb):477
	from :0
```

**Comment by mbryzek** [comment]
code reorganized; last release contained much simpler to follow methods
